Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteProjectController.java

```diff
@@ -215,5 +248,55 @@ public String visualizeAllNetworks(@PathVariable("projectUnixName") String proje
         return "sites/networks/explore";
 
     }
+    
+    /**
+     * This method displays the recent project blog
+     * 
+     * If the project has been set to 'accessible', then the public website page
+     * is displayed. If the project does not exist then an error page is shown.
+     * 
+     * @param unixName
+     *            unix name that is given to the project at the time of its
+     *            creation
+     * @param model
+     *            Model object to map values to view
+     * @return returns a string to access the external website main page
+     * @throws QuadrigaStorageException
+     *             Database storage exception thrown
+     */
+    @RequestMapping(value = "sites/{ProjectUnixName}/projectblogdetails", method = RequestMethod.GET)
+    public String projectblogdetails(Model model, @PathVariable("ProjectUnixName") String unixName, Principal principal,
+            @InjectProject(unixNameParameter = "ProjectUnixName") IProject project) throws QuadrigaStorageException {
+
+        String user = null;
+        if (principal != null) {
+            user = principal.getName();
+        }
+        // Fetch blog entries for a project identified by project unix name
+        String projectId = project.getProjectId();
+        Integer count = 1;
+        List<IProjectBlogEntry> projectBlogEntryList = projectBlogEntryManager.getProjectBlogEntryList(projectId,
+                count);
+        
+        if (user == null) {
+            if (projectManager.getPublicProjectWebsiteAccessibility(unixName)) {
+                model.addAttribute("projectBlogEntryList", projectBlogEntryList);
+                model.addAttribute("project", project);
+                return "sites/projectblogdetails";
+            } else {
+                return "forbidden";
+            }
+        }
+
+        if (projectManager.getPrivateProjectWebsiteAccessibility(unixName, user)
```

Comment:
Merge this if statement with the previous one. It's mostly code duplication besides the if statements conditions.